include/roaring/roaring.h

```diff
@@ -94,8 +95,25 @@ void roaring_bitmap_printf_describe(const roaring_bitmap_t *r);
 
 /**
  * Creates a new bitmap from a list of uint32_t integers
+ *
+ * This function is deprecated, use `roaring_bitmap_from` instead, which
+ * doesn't require the number of elements to be passed in.
+ *
+ * @see roaring_bitmap_from
  */
-roaring_bitmap_t *roaring_bitmap_of(size_t n, ...);
+CROARING_DEPRECATED roaring_bitmap_t *roaring_bitmap_of(size_t n, ...);
+
+/**
+ * Creates a new bitmap which contains all values passed in as arguments.
+ *
+ * @see roaring_bitmap_of_ptr
+ */
+// While __VA_ARGS__ occurs twice in expansion, one of the times is in a sizeof expression, which is an
+// unevaluated context, so it's even safe in the case where expressions passed have side effects
+// (roaring_bitmap_from(my_func(), ++i))
+#define roaring_bitmap_from(...) \
```

Comment:
This needs to be guarded so that it builds only when compiling as a C program.

Reply:
In C++, you need different code: 

https://news.ycombinator.com/item?id=11078288